src/test/harness.rs

```diff
@@ -342,3 +358,25 @@ fn char_of_byte_index(s: &str, byte: usize) -> usize {
 
     panic!("Couldn't find byte {} in {:?}", byte, s);
 }
+
+pub fn test_data_to_tmp_dir<P: AsRef<Path>>(test_data_path: P) -> Result<TempDir, Box<Error>> {
+    let cur_dir = {
+        use build::environment::{EnvironmentLock, Environment};
+        let env = EnvironmentLock::get();
+        let (guard, _other) = env.lock();
+        Environment::push_with_lock(&HashMap::new(), None, guard)
```

Comment:
I'd rather not duplicate this bit - it seems dodgy in `new`, in the first place.